tests/python/pants_test/engine/legacy/test_options_parsing.py

```diff
@@ -0,0 +1,51 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import unittest
+from contextlib import contextmanager
+
+from pants.base.build_environment import get_buildroot
+from pants.engine.legacy.options_parsing import Options, OptionsParseRequest
+from pants.init.engine_initializer import EngineInitializer
+from pants_test.engine.util import init_native
+
+
+class TestEngineOptionsParsing(unittest.TestCase):
+
+  _native = init_native()
+
+  @contextmanager
```

Comment:
Seems a little odd that this is a context manager, rather than just a helper function... Is there a reason for it being a context manager?

Reply:
my thinking is that this will likely expand to setup other aspects of tests, like temp dirs and such which will almost certainly eventually require contextmanagers (and I'd ultimately like to refactor all of the engine-related tests to a common base test class for reuse where contextmanagers are heavily used already).